src/components/RoomDropdown/index.tsx

```diff
@@ -0,0 +1,216 @@
+    display: flex;
+    td {
+        line-height: 13px;
+        &:last-child {
+            margin-left: 15px;
+        }
+    }
+    &:first-child {
+        padding-top: 15px;
+        border-radius: 10px 10px 0 0;
+    }
+    &:last-child {
+        padding-bottom: 15px;
+        border-radius: 0 0 10px 10px;
+    }
+    :hover {
+        background-color: #f1f4f7;
+    }
+`
+
+const RoomDropdown = ({ button }: Props) => {
+    const [isOpen, setIsOpen] = useState(false)
+    const ref = useRef<HTMLDivElement>(null)
+
+    useEffect(() => {
+        /**
+         * Alert if clicked on outside of element
+         */
+        function handleClickOutside(event: any) {
+            if (ref.current && !ref.current.contains(event.target)) {
+                setIsOpen(false)
+            }
+        }
+
+        // Bind the event listener
+        document.addEventListener('mousedown', handleClickOutside)
+        return () => {
+            // Unbind the event listener on clean up
+            document.removeEventListener('mousedown', handleClickOutside)
+        }
+    }, [ref])
+
+    const toggleOpen = () => {
+        setIsOpen(!isOpen)
+    }
+
+    return (
+        <DropDownContainer>
+            {React.cloneElement(button, { onClick: toggleOpen }) || (
+                <DropDownHeader onClick={toggleOpen}>Mangoes</DropDownHeader>
+            )}
+            {isOpen && (
+                <DropDownListContainer ref={ref}>
+                    <DropDownList>
+                        <ListItem>
+                            <td>
+                                <svg
+                                    width="16"
+                                    height="16"
+                                    viewBox="0 0 16 16"
+                                    fill="none"
+                                    xmlns="http://www.w3.org/2000/svg"
+                                >
+                                    <path
+                                        d="M6.5 2C4.84315 2 3.5 3.34315 3.5 5C3.5 6.65685 4.84315 8 6.5 8C8.15685 8 9.5 6.65685 9.5 5C9.5 3.34315 8.15685 2 6.5 2ZM2 5C2 2.51472 4.01472 0.5 6.5 0.5C8.98528 0.5 11 2.51472 11 5C11 7.48528 8.98528 9.5 6.5 9.5C4.01472 9.5 2 7.48528 2 5ZM13.25 7.25C13.6642 7.25 14 7.58579 14 8V8.75H14.75C15.1642 8.75 15.5 9.08579 15.5 9.5C15.5 9.91421 15.1642 10.25 14.75 10.25H14V11C14 11.4142 13.6642 11.75 13.25 11.75C12.8358 11.75 12.5 11.4142 12.5 11V10.25H11.75C11.3358 10.25 11 9.91421 11 9.5C11 9.08579 11.3358 8.75 11.75 8.75H12.5V8C12.5 7.58579 12.8358 7.25 13.25 7.25ZM3.875 12.5C2.93041 12.5 2 13.4102 2 14.75C2 15.1642 1.66421 15.5 1.25 15.5C0.835786 15.5 0.5 15.1642 0.5 14.75C0.5 12.7761 1.9201 11 3.875 11H9.125C11.0799 11 12.5 12.7761 12.5 14.75C12.5 15.1642 12.1642 15.5 11.75 15.5C11.3358 15.5 11 15.1642 11 14.75C11 13.4102 10.0696 12.5 9.125 12.5H3.875Z"
+                                        fill="#59799C"
+                                    />
+                                </svg>
```

Comment:
I'm trying to exercise a different practice with the icons currently. Few rules:
- if an `svg` has a dynamic (depending on state) element, I bake the SVG code into the react code,
- if an `svg` does not have any dynamic element, or color, or background, or whatever I try to separate such thing in an svg file.

Maybe that's something for you to consider.

SVGs tend to polute the code with them lengthty path lines.

For an example see `ChatWindow` and `ModifyIcon`, and `MoreIcon`.

Reply:
I've made the icons into components which are all exported from a single folder in 5bdcb325692e5fd287421a475ab1c6ff5df18e27. It's a convention I've used before, would this be okay? I would be fine with changing it as well.